### PR TITLE
Added a sample data that simulates electron beam shape

### PR DESCRIPTION
**Pull request by yijiang1**
Added a sample data that simulates electron beam shape.
The default parameters generate a typical electron beam shape.

### CODE REVIEW

**Comment by Hovden** [comment]
@mathturtle Shawn, can you review this pull request?

File: tomviz/PythonGeneratedDatasetReaction.cxx [hunk 1]
```diff
@@ -370,10 +370,11 @@ void PythonGeneratedDatasetReaction::addDataset()
     layout->addWidget(buttons);
     dialog.setLayout(layout);
     dialog.layout()->setSizeConstraint(QLayout::SetFixedSize); //Make the UI non-resizeable
-
+
```
Comment (mathturtle) [inline tomviz/PythonGeneratedDatasetReaction.cxx]: Blank lines should be empty with no spaces.  (I can fix this after you merge if you want though).

Comment (mathturtle) [inline tomviz/PythonGeneratedDatasetReaction.cxx]: Blank lines should be empty with no spaces.  (I can fix this after you merge if you want though).

Comment (mathturtle) [inline tomviz/PythonGeneratedDatasetReaction.cxx]: Blank lines should be empty with no spaces.  (I can fix this after you merge if you want though).

**Comment by mathturtle** [comment]
If I set the x-y pixel size to 0, I get a ZeroDivisionError from python.

File: tomviz/PythonGeneratedDatasetReaction.cxx [hunk 2]
```diff
@@ -384,8 +385,175 @@ void PythonGeneratedDatasetReaction::addDataset()
...
+      pythonScript.replace("###f_a3###", QString("f_a3 = %1").arg(f_a3->value()));
+      pythonScript.replace("###phi_a3###", QString("phi_a3 = %1").arg(phi_a3->value()));
+      pythonScript.replace("###f_c3###", QString("f_c3 = %1").arg(f_c3->value()));
+      pythonScript.replace("###phi_c3###", QString("phi_c3 = %1").arg(phi_c3->value()));
+      generator.setScript(pythonScript);
+      const int shape[3] = {Nxy->value(),Nxy->value(),Nz->value()};
+      this->dataSourceAdded(generator.createDataSource(shape));
+    }
+  
+    } //end of else if
```
Comment (mathturtle) [inline tomviz/PythonGeneratedDatasetReaction.cxx]: This should be indented only 2 spaces like the opening of this block.  I can fix the indentation stuff though.

File: tomviz/PythonGeneratedDatasetReaction.cxx (hunk 1 above)
Comment (mathturtle) [inline tomviz/PythonGeneratedDatasetReaction.cxx]: Ok, apparently I found a bug in Github... It told me it failed to comment on this the first few times.

**Comment by mathturtle** [comment]
The divide by 0 thing it the only thing that I would say prevents this from being merged.  I can fix up the indentation for you afterwards.  I don't know the science well enough to say if what it is doing is right... but It doesn't break and the C++ side of the code makes sense.  I can read over the python too, but that may take a while if you want me to understand it.

**Comment by yijiang1** [comment]
Thanks. Github is behaving very weird. I'm working on the bugs and style
now.

**Comment by yijiang1** [comment]
@mathturtle I've deleted empty spaces and fixed the dividing-by-zero bug.

**Pull request merged**